generate-analytical-dataset.py

```diff
@@ -27,35 +27,38 @@ def main():
         .enableHiveSupport()
         .getOrCreate()
     )
-    adg_hive_table = "analytical_dataset_generation_staging.core_contract_hbase"
-    adg_hive_select_query = "select * from %s" % adg_hive_table
-    df = spark.sql(adg_hive_select_query)
-    keys_map = {}
-    values = (
-        df.select("data")
-        .rdd.map(lambda x: getTuple(x.data))
-        .map(lambda y: decrypt(y[0], y[1], y[2], y[3], keys_map))
-    )
-    row = Row("val")
-    datadf = values.map(row).toDF()
-    datadf.show()
-    adg_parquet_name = "core_contract.parquet"
-    parquet_location = "s3://%s/analytical-dataset/%s" % (
-        S3_PUBLISH_BUCKET,
-        adg_parquet_name,
-    )
-    datadf.write.mode("overwrite").parquet(parquet_location)
-    src_hive_table = "analytical_dataset_generation.core_contract"
-    src_hive_drop_query = "DROP TABLE IF EXISTS %s" % src_hive_table
-    src_hive_create_query = (
-        """CREATE EXTERNAL TABLE IF NOT EXISTS %s(val STRING) STORED AS PARQUET LOCATION "%s" """
-        % (src_hive_table, parquet_location)
-    )
-    spark.sql(src_hive_drop_query)
-    spark.sql(src_hive_create_query)
-    src_hive_select_query = "select * from %s" % src_hive_table
-    spark.sql(src_hive_select_query).show()
 
+    database_name = "analytical_dataset_generation_staging"
+    tables = getTables(database_name)
+    for table_to_process in tables:   
+        adg_hive_table = database_name + "." + table_to_process 
+        adg_hive_select_query = "select * from %s" % adg_hive_table
+        df = spark.sql(adg_hive_select_query)
+        keys_map = {}
+        values = (
+            df.select("data")
+            .rdd.map(lambda x: getTuple(x.data))
+            .map(lambda y: decrypt(y[0], y[1], y[2], y[3], keys_map))
+        )
+        row = Row("val")
+        datadf = values.map(row).toDF()
+        datadf.show()
+        adg_parquet_name = "core_contract.parquet"
+        parquet_location = "s3://%s/analytical-dataset/%s" % (
+            S3_PUBLISH_BUCKET,
+            adg_parquet_name,
+        )
+        datadf.write.mode("overwrite").parquet(parquet_location)
+        src_hive_table = "analytical_dataset_generation.core_contract"
```

Comment:
maybe the analytical_dataset_generation glue table should be declared as a database name as you have done for the staging one? 
database_name_staging="analytical_dataset_generation_staging"
database_name_published="analytical_dataset_generation"